sql/2020/21_Resource_Hints/21_15.sql

```diff
@@ -0,0 +1,39 @@
+#standardSQL
+#19_15: Count of preload HTTP Headers with nopush attribute set.
+CREATE TEMPORARY FUNCTION getLinkHeaders
+(payload STRING)
+RETURNS ARRAY<STRING>
+LANGUAGE js AS """
+  var $ = JSON.parse(payload);
+  var headers = $.response.headers;
+  var preload=[];
+  
+  for (i in headers) {
+      if (headers[i].name.toLowerCase() === 'link')
+        preload.push(headers[i].value);
+      }
+     return preload;  
+    
+""";
+
+SELECT
+    client,
+    firstHtml,
```

Comment:
Why are you grouping and selecting `firstHtml` here. Can you group by `page` instead?

Reply:
This will differentiate documents that preload/nopush from secondary 1P or 3P resources, I think this is ok.